src/core/ucc_context.c

```diff
@@ -544,10 +541,27 @@ ucc_status_t ucc_core_addr_exchange(ucc_context_t          *context,
                     ucc_error("proc info collision: %d %d", r, i);
                     return UCC_ERR_NO_MESSAGE;
                 }
-                r = (ucc_rank_t)i;
+                r = i;
             }
         }
         addr_storage->rank = r;
+        /*check if TLs arrays is symmetric*/
+        addr_storage->flags = UCC_ADDR_STORAGE_FLAG_TLS_SYMMETRIC;
+        h = UCC_ADDR_STORAGE_RANK_HEADER(addr_storage, 0);
```

Comment:
do you really need to iterate over addr_storage 2nd time? could you just do the same in the loop above (adding if (i > 1)) ?

Reply:
merged two loops